compiler/codegen/OMRCodeGenerator.hpp

```diff
@@ -386,7 +386,40 @@ class OMR_EXTENSIBLE CodeGenerator
 
    TR_HasRandomGenerator randomizer;
 
-   bool supportsAtomicAdd() {return false;}
+   /** \brief
+    *     Determines whether intrinsics on the following atomic symbols is supported:
+    *
+    *     - atomicAdd32BitSymbol
+    *     - atomicAdd64BitSymbol
+    *     - atomicFetchAndAdd32BitSymbol
+    *     - atomicFetchAndAdd64BitSymbol
+    *
+    *  \return
+    *     \c true if intrinsics are supported; \c false otherwise.
+    */
+   bool supportsAtomicAdd()
```

Comment:
I don't whether this query should cover support for all above 4 symbols. There may be a processor that supports one but not the other, and hence the codegen may choose to support one but not the other. Though I can't think of any current example, but the original 8086 processor supports only atomic add but not atomic fetch and add.

Why don't we make it a generic query like `bool supportsNonHelper(CommonNonhelperSymbol symbol)` so that support for each individual symbol can be queried?

Reply:
I definitely agree. That's a great idea. I'm working on this now.

Reply:
Fixed in a7704d1